metomi/rose/tests/test_loc_handlers_rsync_remote_check.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (C) British Crown (Met Office) & Contributors.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+import sys
+
+from ast import literal_eval
+
+from metomi.rose.loc_handlers.rsync_remote_check import main
+
+
+def test_check_file(monkeypatch, capsys, tmp_path):
+    content = 'blah'
+    permission_level = '0o100777'
```

Comment:
Not sure what the leading `100` does?

Not keen on creating 777 files! Can we downgrade to 700, 644 or something less scary.

Reply:
Totally. In one case 644 was fine, in another 700 was as low as I could go without upsetting Pytest's cleanup.